src/object.c

```diff
@@ -1236,6 +1236,11 @@ struct redisMemOverhead *getMemoryOverheadData(void) {
         mh->db[mh->num_dbs].overhead_ht_expires = mem;
         mem_total+=mem;
 
+        // Account for the slot to keys map in cluster mode
+        mem = dictSize(db->dict) * dictMetadataSize(db->dict);
```

Comment:
This will be an easy point to miss if we ever add more dictionary metadata. I would suggest just doing:
```suggestion
        mem = dictSize(db->dict) * sizeof(clusterDictEntryMetadata);
```

Reply:
Agree

Reply:
@madolson i didn't understand that argument. i think it could go the other way around.
i.e. `dictMetadataSize` is what's truly being used, and `sizeof(clusterDictEntryMetadata)` could go outdated some day.
also, looking at the diff, i like the previous version better.

Reply:
@oranagra If we add more metadata, it might not have anything to do with the cluster data though? I would guess we are far more likely to add non-cluster metadata compared to cluster metadata.

Reply:
ohh, i got your point.
so i still prefer the old approach.
the reason is that if we forget to update that code, this overhead will still be correctly reported somewhere, and we would be more easily be able to dig and find what we missed.
if we use the new approach, and we forget, we'll just think of that additional data ad values.

also the old code looks better.

Reply:
@oranagra how about we add an assert they are the same size here with a comment. So if anything changes we'll notice? I think we can do static asserts as part of the build since we moved to C11. Then we can use the code you find nicer looking.

Reply:
we didn't move to C11 (we moved back :smile: )
we can add a normal assert, as long as it's in a commonly used path (INFO is ok, MEMORY isn't).
or we can just add a bold comment in the place that configures the dict, and refer here.

Reply:
Apparently AWS didn't get the message, we stayed on C11. I'd vote for an assert in info. A comment here isn't that useful since we won't think about this code when adding metadata.

Reply:
i didn't mean a comment here.
i meant a comment in:
```c
size_t dictEntryMetadataSize(dict *d) {
    UNUSED(d);
    return server.cluster_enabled ? sizeof(clusterDictEntryMetadata) : 0;
}
```

Reply:
Oh, I see. I still have a preference for an assert, but a comment there is likely sufficient.